data/crac/crac-impl/src/main/java/com/powsybl/openrao/data/cracimpl/AbstractRemedialAction.java

```diff
@@ -135,16 +125,16 @@ private void updateMapWithValue(Map<State, UsageMethod> map, State key, UsageMet
     public Set<FlowCnec> getFlowCnecsConstrainingUsageRules(Set<FlowCnec> perimeterCnecs, Network network, State optimizedState) {
         Set<FlowCnec> toBeConsideredCnecs = new HashSet<>();
         Set<UsageRule> usageRulesOnFlowConstraint = new HashSet<>();
-        usageRulesOnFlowConstraint.addAll(getUsageRules(OnFlowConstraint.class, optimizedState));
+        usageRulesOnFlowConstraint.addAll(getUsageRules(OnConstraint.class, optimizedState).stream().filter(onConstraint -> onConstraint.getCnec() instanceof FlowCnec).toList());
         usageRulesOnFlowConstraint.addAll(getUsageRules(OnFlowConstraintInCountry.class, optimizedState));
         usageRulesOnFlowConstraint.forEach(usageRule -> toBeConsideredCnecs.addAll(getFlowCnecsConstrainingForOneUsageRule(usageRule, perimeterCnecs, network)));
         return toBeConsideredCnecs;
     }
 
     // TODO: move this method to RaoUtil
     public Set<FlowCnec> getFlowCnecsConstrainingForOneUsageRule(UsageRule usageRule, Set<FlowCnec> perimeterCnecs, Network network) {
-        if (usageRule instanceof OnFlowConstraint onFlowConstraint) {
-            return Set.of(onFlowConstraint.getFlowCnec());
+        if (usageRule instanceof OnConstraint<?> onConstraint) {
+            return onConstraint.getCnec() instanceof FlowCnec ? Set.of((FlowCnec) onConstraint.getCnec()) : Set.of();
```

Comment:
This isn't quite the same as the old behaviour (for instance if we give an angleCnec, it will return an empty set instead of throwing). I would suggest this
```suggestion
        if (usageRule instanceof OnConstraint<?> onConstraint && onConstraint.get(Cnec()) instanceof FlowCnec flowCnec) {
            return Set.of(flowCnec);
```